ludwig/utils/image_utils.py

```diff
@@ -163,68 +163,110 @@ def read_image_from_str(img: str, num_channels: Optional[int] = None) -> torch.T
 
 def pad(
     img: torch.Tensor,
-    size: Union[int, Tuple[int]],
+    size: Union[int, Tuple[int, int]],
 ) -> torch.Tensor:
-    old_size = np.array(img.shape[1:])
-    pad_size = (to_np_tuple(size) - old_size) / 2
-    padding = np.concatenate((np.floor(pad_size), np.ceil(pad_size)))
+    """torchscript-compatible implementation of pad.
+
+    Args:
+        img (torch.Tensor): image with shape [..., height, width] to pad
+        size (Union[int, Tuple[int, int]]): size to pad to. If int, resizes to square image of that size.
+
+    Returns:
+        torch.Tensor: padded image of size [..., size[0], size[1]] or [..., size, size] if size is int.
+    """
+    if torch.jit.isinstance(size, int):
```

Comment:
Since this conditional seems to be common across all preprocessing methods, could we abstract this out? Not sure if that affects torchscript compatibility.

Reply:
I think we probably could! Depends somewhat on the use of `pad` and `crop`. I assumed that because they were public functions, they could conceivably be called independently from one another.

On the other hand, if the only entry point into these functions is `resize_image`, then I believe we could abstract it out so the check only happens once in `resize_image`. What do you think?

Reply:
Maybe a utility function like `to_tensor_pair` whose implementation is these four lines?

Reply:
Ah, I see what you were saying now. Made those changes. Thanks!